### PR TITLE
COMP110 Worksheet 6 Submission - Aleksandar Hristov

### PR DESCRIPTION
**Pull request by AH228589**
Implemented methods for a match 3 game.

### CODE REVIEW

**Comment by AH228589** [comment]
Added regular expression to make sure the input is always in the correct format.
Fixed outside bounds of array.
Added checks for diagonals.

File: comp110-worksheet-6/OxoBoard.cs [hunk 1]
```diff
@@ -3,74 +3,149 @@
...
+        {
+            return gameBoard[x, y];
+        }
+
+        // If the specified square is currently empty, fill it with mark and return true.
+        // If the square is not empty, leave it as-is and return False.
+        public bool SetSquare(int x, int y, Mark mark)
+        {
+            string inputString = x + "," + y;
+            if (format.IsMatch(inputString))
```
Comment (edpowley) [inline comp110-worksheet-6/OxoBoard.cs:48]: Constructing a string and checking it against a regex is overkill here -- just checking that both x and y are in range would be more efficient. (The regex check would make more sense used directly on the input string from the user)

File: comp110-worksheet-6/OxoBoard.cs [hunk 2]
```diff
@@ -3,74 +3,149 @@
...
+        public Mark GetWinner()
+        {
+            for (int i = 0; i < width; i++)
+            {
+                for (int j = 0; j < height; j++)
+                {
+                    if (gameBoard[i, j] != Mark.None) //Only look inside the cell if it's not empty.
+                    {
+                        int currI = i + 2, currJ = j + 2;
+                        int[] currSquare = new int[] { i, j };
```
Comment (edpowley) [inline comp110-worksheet-6/OxoBoard.cs:94]: Are these variables actually used?

File: comp110-worksheet-6/OxoBoard.cs [hunk 3]
```diff
@@ -3,74 +3,149 @@
...
+                        if (currI < 3 && currJ < 3)
```
Comment (edpowley) [inline comp110-worksheet-6/OxoBoard.cs:95]: I suspect this check is what's causing some of the unit tests to fail -- some of the tests are being skipped

**Review by edpowley — COMMENTED** [review]
A few unit tests failing, but otherwise a decent implementation. More comments would be good to aid readability.

**Pull request closed without merging**